TestFoundation/TestNSURLSession.swift

```diff
@@ -164,7 +259,13 @@ class TestURLSession : XCTestCase {
     }
 }
 
-class DataTask: NSObject {
+#if os(Linux)
+typealias FoundationObject = NSObject
+#else
+typealias FoundationObject = SwiftFoundation.NSObject
+#endif
```

Comment:
I'd like to understand more about the root cause of this error. Clients of Foundation should absolutely be able to subclass NSObject without this kind of hackery.